src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -0,0 +1,317 @@
+            + PREFIX_SERVING_SIZE + "SERVING SIZE\n"
+            + "[" + PREFIX_INGREDIENT + "INGREDIENT]...\n"
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_STEP + "STEP]...\n\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INDEX + "1 "
+            + PREFIX_NAME + "Aglio Olio "
+            + PREFIX_COMPLETION_TIME + "10 "
+            + PREFIX_SERVING_SIZE + "1 "
+            + PREFIX_INGREDIENT + "Spaghetti "
+            + PREFIX_INGREDIENT + "Garlic "
+            + PREFIX_INGREDIENT + "Sauce "
+            + PREFIX_INGREDIENT + "Bacon "
+            + PREFIX_STEP + "Cook the pasta "
+            + PREFIX_STEP + "Saute the garlic "
+            + PREFIX_STEP + "Toss pasta in the sauce "
+            + PREFIX_STEP + "Add bacon in pasta "
+            + PREFIX_STEP + "Taste and season "
+            + PREFIX_TAG + "Italian "
+            + PREFIX_TAG + "Western";
+
+    public static final String MESSAGE_EDIT_RECIPE_SUCCESS = "Edited Recipe: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_RECIPE = "This recipe already exists in the recipe book.";
+
+    private Name targetName;
+    private Index targetIndex;
+    private final EditRecipeDescriptor editRecipeDescriptor;
+
+    /**
+     * @param name of the recipe in the filtered recipe list to edit
+     * @param editRecipeDescriptor details to edit the recipe with
+     */
+    public EditCommand(Name name, EditRecipeDescriptor editRecipeDescriptor) {
+        requireNonNull(name);
+        requireNonNull(editRecipeDescriptor);
+
+        this.targetName = name;
+        this.editRecipeDescriptor = new EditRecipeDescriptor(editRecipeDescriptor);
+    }
+
+    /**
+     * @param index of the recipe in the filtered recipe list to edit
+     * @param editRecipeDescriptor details to edit the recipe with
+     */
+    public EditCommand(Index index, EditRecipeDescriptor editRecipeDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editRecipeDescriptor);
+
+        this.targetIndex = index;
+        this.editRecipeDescriptor = new EditRecipeDescriptor(editRecipeDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Recipe> lastShownList = model.getFilteredRecipeList();
+
+        // Guaranteed that either targetIndex or targetName is non-null.
+        Recipe recipeToEdit = (targetIndex != null)
+                ? getRecipe(lastShownList, targetIndex)
+                : getRecipe(lastShownList, targetName);
+
+        Recipe editedRecipe = createEditedRecipe(recipeToEdit, editRecipeDescriptor);
+
+        if (!recipeToEdit.isSameRecipe(editedRecipe) && model.hasRecipe(editedRecipe)) {
+            throw new CommandException(MESSAGE_DUPLICATE_RECIPE);
+        }
```

Comment:
`model.hasRecipe(editedRecipe)` is case-sensitive so an edit of "taco" when there is an existing recipe called "TACO" will not be caught here and will be allowed to execute. This is incorrect behaviour as view and delete commands are not case-sensitive and should not expect both "TACO" and "taco" to be in the recipe book at the same time.

Reply:
I just realised that this issue is also present in add command and should be addressed too 😱

Reply:
Changes have been made to ignore case sensitivity, updated Name.equals method